scripts/cmake/vcpkg_download_distfile.cmake

```
@@ -80,6 +80,11 @@ function(vcpkg_download_distfile VAR)
             message(FATAL_ERROR "vcpkg_download_distfile must not be passed both SHA512 and SKIP_SHA512.")
         endif()
     endif()
+    string(LENGTH "${vcpkg_download_distfile_SHA512}" vcpkg_download_distfile_SHA512_length)
+    if(vcpkg_download_distfile_SHA512_length LESS 128)
+        # Pad with zeroes to make a valid SHA512
+        string(SUBSTRING "${vcpkg_download_distfile_SHA512}00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000" 0 128 vcpkg_download_distfile_SHA512)
+    endif()
```

```suggestion
    if(vcpkg_download_distfile_SHA512 STREQUAL "0")
        string(REPEAT "0" 128 vcpkg_download_distfile_SHA512)
    endif()
    string(LENGTH "${vcpkg_download_distfile_SHA512}" vcpkg_download_distfile_SHA512_length)
    if(NOT vcpkg_download_distfile_SHA512_length EQUAL 128)
        message(FATAL_ERROR "Invalid SHA512: ${vcpkg_download_distfile_SHA512}")
    endif()
```

can we do something like this instead?

---

or maybe:
```suggestion
    string(LENGTH "${vcpkg_download_distfile_SHA512}" vcpkg_download_distfile_SHA512_length)
    if(NOT vcpkg_download_distfile_SHA512_length EQUAL "128")
            string(REPEAT "0" 128 vcpkg_download_distfile_SHA512)
    endif()
```